### PR TITLE
CLEAN: adding a few lines of code; cleaning some code;

### PR DESCRIPTION
**Pull request by b-elfo**

### CODE REVIEW

File: downstream.py [hunk 1]
```diff
@@ -206,13 +207,7 @@ def validate_task(dataset_path: str,
 
     idx_to_class = {v:k for k,v in dataset.class_to_idx.items()}
 
-    valid_size = dataset.__len__()
-    data_split = [dataset.__len__()-valid_size, valid_size]
-    
-    train_dataset, valid_dataset = torch.utils.data.random_split(dataset, 
-                                                                 lengths=data_split)
-
-    valid_dataloader = DataLoader(dataset=valid_dataset,
```
Comment (b-elfo) [inline downstream.py:215]: entire dataset is used for validation; this will change soon;

File: downstream.py [hunk 2]
```diff
@@ -243,22 +235,32 @@ def validate_task(dataset_path: str,
...
+
+            if len(samples) < num_samples:
+                samples.append([input,label_idx,predict_idx])
+    
+    fig_num = int(num_samples**(0.5))
+    fig, axs = plt.subplots(fig_num,
+                            fig_num,
+                            constrained_layout=True,
+                            )
+    fig.suptitle(f'Accuracy: {correct_num/total_num}', fontsize=16)
```
Comment (b-elfo) [inline downstream.py:253]: Logging accuracy to validation;

File: downstream.py [hunk 3]
```diff
@@ -243,22 +235,32 @@ def validate_task(dataset_path: str,
...
-    fig, axs = plt.subplots(3,3)
     i,j = 0,0
     for (input, label, predict) in samples:
-        label_idx = label.item()
-        predict_idx = torch.argmax(predict).item()
         axs[j][i].imshow(input[0].cpu().permute(1,2,0))
-        axs[j][i].set_title(f'{idx_to_class[label_idx]} - {idx_to_class[predict_idx]}')
-        if label_idx != predict_idx:
+        axs[j][i].set_title(f'{idx_to_class[label]} - {idx_to_class[predict]}')
+        if label != predict:
```
Comment (b-elfo) [inline downstream.py:259]: Cleaning code;

File: downstream.py [hunk 4]
```diff
@@ -243,22 +235,32 @@ def validate_task(dataset_path: str,
...
             rect = patches.Rectangle((0, 0), 31, 31, linewidth=3, edgecolor='r', facecolor='none')
             axs[j][i].add_patch(rect)
         i += 1
-        if i >= 3:
+        if i >= fig_num:
```
Comment (b-elfo) [inline downstream.py:263]: cleaning;
